internal/oci/pull.go

```diff
@@ -141,11 +142,19 @@ type RemoteClient struct{}
 // Image fetches an image manifest. The returned image lazily pulls its layers.
 func (i *RemoteClient) Image(ctx context.Context, ref name.Reference, o ...ImageClientOption) (ociv1.Image, error) {
 	opts := parse(o...)
+	if opts.pull == ImagePullPolicyNever {
+		return nil, errors.New(errPullNever)
+	}
 	if opts.auth != nil {
 		return remote.Image(ref, remote.WithContext(ctx), remote.WithAuth(opts.auth))
 	}
-	if opts.pull == ImagePullPolicyNever {
-		return nil, errors.New(errPullNever)
+	keychain, err := k8schain.NewInCluster(ctx, k8schain.Options{})
```

Comment:
Will `NewInCluster` always fail if in-cluster auth isn't supported?

Or more generally, what causes `NewInCluster` to fail and what causes `remote.WithAuthFromKeychain` to fail? It may be worth adding comments to clarify.